magento/magento2#13765 Excess requests 'customer data' on checkout cart page were fixed

andrewbess opened the pull request:
### Description
The customer/section/load gets called 4 times when loading the cart for the first time. 3 of those times, cart data gets returned. On Magento 2.1, I believe only two calls are made to customer/section/load.

### Fixed Issues (if relevant)
1. magento/magento2#13765: "cart" section data gets loaded 3 times on cart page (2.2.2)

### Steps to reproduce
1. Add any product to cart
2. Go to cart page `/checkout/cart` with the "Network" panel open in DevTools.
### Expected result
1. The `/customer/section/load` url should only get called twice. For example, these are the two urls that get called on a Magento 2.1 install:
```
https://magento22.dev/customer/section/load/?sections=&update_section_id=false&_=1519190370042
https://magento22.dev/customer/section/load/?sections=directory-data&update_section_id=false&_=1519190370043
```

### Actual result
1. The `/customer/section/load` url will get called four times:
```
https://magento22.dev/customer/section/load/?sections=&update_section_id=false&_=1519190370042
https://magento22.dev/customer/section/load/?sections=directory-data&update_section_id=false&_=1519190370043
https://magento22.dev/customer/section/load/?sections=cart&update_section_id=false&_=1519190370045
https://magento22.dev/customer/section/load/?sections=cart&update_section_id=false&_=1519190370046
```

Note: if you refresh the cart, there will only be two calls made to `/customer/section/load`.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
magento/magento2#13765 Excess requests 'customer data' on checkout cart page were fixed:

- Excess request for 'directory-data' was fixed;
- Excess requests for 'cart' were fixed.

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=14314) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=14314) before we can accept your contribution.<br/>

Commit:
magento/magento2#13765 "cart" section data gets loaded 3 times on cart page (2.2.2):

- Blank line after variable declarations was added;
- Blank line before "if" was added.

ihor-sviziev commented on app/code/Magento/Checkout/view/frontend/web/js/view/minicart.js:
```diff
@@ -94,17 +94,17 @@ define([
                 sidebarInitialized = false;
                 this.update(updatedCart);
+
+                if (cartData()['website_id'] !== window.checkout.websiteId) {
```
AFAIK this file is using not at checkout page only, so window.checkout will not be available

andrewbess replied:
@ihor-sviziev 
I checked your comment.
window.checkout is available on home page and category pages.
![2018-03-26 09-51-13](https://user-images.githubusercontent.com/10788564/37891075-9fcdf9c0-30db-11e8-9d4f-ed470e4525ca.png)
Also, I want to tell you, that logic "cartData()['website_id'] !== window.checkout.websiteId" was earlier in Magento.
It was fixed it to remove excessed request.

ihor-sviziev replied:
Will review deeper little bit later

sidolov replied:
I can confirm that window.checkout property exist at any page where present minicart

andrewbess replied:
Hello, @sidolov 
Please check and approve this PR.
Thanks for advance.

andrewbess commented:
It was fixed.

ihor-sviziev commented:
@magento-engcom-team please prepare new test environment

ihor-sviziev commented:
@magento-engcom-team give me new test instance

magento-engcom-team commented:
Hi @ihor-sviziev, here is your [new test Magento instance](http://pr-14314.engcom.dev.magento.com)

magento-engcom-team commented:
Hi @andrewbess. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.2.5 release.

andrewbess commented:
Hello @xmav 
Thank you.
I will check it.

ihor-sviziev commented:
@xmav there also was forwardport to 2.3-develop https://github.com/magento/magento2/pull/14782. Was it also reverted?

xmav commented:
@ihor-sviziev yes, it was reverted.

sshymko commented:
@xmav 
Is there an alternative fix to be used instead of the reverted one?

elvinristi commented:
Problem with this PR is that it probably works only then when you have to show prices excluding tax.

when I update to from "`quote.totals().subtotal`" to "`quote.totals().subtotal_incl_tax`" then this works with inclusive tax.

So there should be added some check what is the admin setting - inclusive or exclusive tax and using proper value based on it.

Ctucker9233 commented:
@xmav Is there an alternate to this fix? It is desperately needed.
